src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -810,9 +827,114 @@ export class StudyViewPageStore {
+        if (!this.clinicalDataBinPromises.hasOwnProperty(uniqueKey)) {
+            this.clinicalDataBinPromises[uniqueKey] = remoteData<DataBin[]>({
+                invoke: () => {
+                    // TODO this.barChartFilters.length > 0 ? 'STATIC' : 'DYNAMIC' (not trivial when multiple filters involved)
+                    return internalClient.fetchClinicalDataBinCountsUsingPOST({
+                        dataBinMethod: DataBinMethodConstants.STATIC,
+                        disableLogScale: this.isLogScaleDisabled(chartMeta.uniqueKey),
+                        attributeId: chartMeta.clinicalAttribute!.clinicalAttributeId,
+                        clinicalDataType: chartMeta.clinicalAttribute!.patientAttribute ? 'PATIENT' : 'SAMPLE',
+                        studyViewFilter: this.filters
+                    });
+                },
+                default: [],
+                onResult: (result) => {
+                    if (!isFiltered(this.userSelections) && result.length < 2) {
+                        this.changeChartVisibility(chartMeta.uniqueKey, false);
+                    }
+                }
+            });
+        }
+        return this.clinicalDataBinPromises[uniqueKey];
+    }
+
+    private async getClinicalDataBySamples(samples: Sample[]) {
+        let clinicalData:{[sampleId:string]: {[attributeId: string]: string}} = {};
+
+        let sampleClinicalData = await defaultClient.fetchClinicalDataUsingPOST({
+            'clinicalDataType': "SAMPLE",
+            'clinicalDataMultiStudyFilter': {
+                'identifiers': _.map(samples, sample => {
+                    return {
+                        entityId: sample.sampleId,
+                        studyId: sample.studyId
+                    }
+                })
+            } as ClinicalDataMultiStudyFilter
+        });
+
+        _.forEach(sampleClinicalData, item => {
+            clinicalData[item.uniqueSampleKey] = { ...(clinicalData[item.uniqueSampleKey] || {}), ['SAMPLE_' + item.clinicalAttributeId]: item.value };
+        })
+
+        let patientClinicalData = await defaultClient.fetchClinicalDataUsingPOST({
+            'clinicalDataType': "PATIENT",
+            'clinicalDataMultiStudyFilter': {
+                'identifiers': _.map(samples, sample => {
+                    return {
+                        entityId: sample.patientId,
+                        studyId: sample.studyId
+                    }
+                })
+            } as ClinicalDataMultiStudyFilter
+        });
+
+        const patientSamplesMap = _.groupBy(samples, sample => sample.uniquePatientKey);
+
+        _.forEach(patientClinicalData, item => {
+            (patientSamplesMap[item.uniquePatientKey] || []).forEach(sample => {
+                clinicalData[sample.uniqueSampleKey] = { ...(clinicalData[sample.uniqueSampleKey] || {}), ['PATIENT_' + item.clinicalAttributeId]: item.value };
+            });
+        });
+        return _.reduce(samples, (acc, sample) => {
```

Comment:
why not just return `clinicalData`

Reply:
Good point, probably missed when moving the function. Done.